assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -65,23 +73,48 @@ const Area = styled.div`
   align-items: center;
   justify-content: center;
   flex-direction: column;
-  padding: 16px 0 24px;
+  ${({ marginBottom }) => marginBottom && `margin-bottom: ${marginBottom}px;`}
 `;
 
-const List = styled(Area)`
+const List = styled(Area).attrs({ as: 'ul', role: 'listbox' })`
   flex-direction: row;
-  align-items: flex-start;
+  align-items: center;
   justify-content: ${({ hasHorizontalOverflow }) =>
     hasHorizontalOverflow ? 'flex-start' : 'center'};
-  overflow-x: ${({ hasHorizontalOverflow }) =>
-    hasHorizontalOverflow ? 'scroll' : 'hidden'};
+  overflow-x: scroll;
+  overflow-y: hidden;
+  margin: 0 0 ${CAROUSEL_BOTTOM_SCROLL_MARGIN}px 0;
+  scrollbar-color: rgba(255, 255, 255, 0.54) transparent;
+  scrollbar-width: auto;
+  &::-webkit-scrollbar {
+    width: ${SCROLLBAR_HEIGHT}px;
+    height: ${SCROLLBAR_HEIGHT}px;
+  }
+  &::-webkit-scrollbar-track {
+    background-color: transparent;
+  }
+  &::-webkit-scrollbar-thumb {
+    background-color: rgba(255, 255, 255, 0.54);
+    border-radius: ${SCROLLBAR_HEIGHT * 2}px;
+  }
 `;
 
 const StyledGridViewButton = styled(GridViewButton)`
   position: absolute;
   bottom: 24px;
 `;
 
+function calculatePageThumbSize(carouselSize) {
+  const aspectRatio = 9 / 16;
+  const availableHeight =
+    carouselSize.height -
+    CAROUSEL_VERTICAL_PADDING * 2 -
+    CAROUSEL_BOTTOM_SCROLL_MARGIN;
+  const height = availableHeight;
+  const width = availableHeight * aspectRatio;
+  return [width, height];
+}
+
 function Carousel() {
```

Comment:
So, in general I have two concerns with this component now:

- It's huge! 200+ SLOC's is too much IMO. Some of the effects and states could be moved to custom hooks.
- The [unconditional inclusion of `<Modal />`](https://github.com/google/web-stories-wp/pull/411/files#diff-9ca2523a045442855dbdd7218d0095acR272-R279) regardless of whether it's displayed or not. Is that really proper practice here? React common best-practice is always `{isOpen && <Modal />}` over `<Modal isOpen={isOpen} />`, but maybe there are special circumstances here?

I know both issues aren't part of this ticket, but if you could look into it, it would be awesome.

Reply:
Makes sense to me! Not sure why I implemented the Modal this way 🤔

Reply:
We should refactor this in a separate PR, makes it easier to review